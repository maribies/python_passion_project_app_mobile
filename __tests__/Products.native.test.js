import React from "react";
import { create } from "react-test-renderer";
import { Products } from "../Components/Products.native";
import products from "../Fixtures/products.json";

jest.useFakeTimers();

test("renders correctly with no data", () => {
  const tree = create(<Products />).toJSON();

  expect(tree).toMatchSnapshot();
  expect.objectContaining({
    Text: "Something went wrong. No products are found yet!",
  });
});

test("renders correctly with undefined data", () => {
  const tree = create(<Products products={undefined} />).toJSON();

  expect(tree).toMatchSnapshot();
  expect.objectContaining({
    Text: "Something went wrong. No products are found yet!",
  });
});

test("renders correctly with data", () => {
  const tree = create(<Products products={products} />).toJSON();

  expect(tree).toMatchSnapshot();
  expect.objectContaining({ Text: "Chanel" });
});
