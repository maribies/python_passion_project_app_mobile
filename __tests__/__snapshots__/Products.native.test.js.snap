// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with data 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
  style={Object {}}
>
  <View>
    <View
      style={Object {}}
    >
      <Image
        source={
          Object {
            "uri": "https://via.placeholder.com/150",
          }
        }
        style={
          Object {
            "height": 223,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "Quattrocento_700Bold",
            "fontSize": 20,
            "fontWeight": "700",
          }
        }
      >
        Chanel
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "FanwoodText_400Regular",
            "fontSize": 20,
          }
        }
      >
        Flap Bag
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginRight": 8,
              "maxWidth": 64,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "marginBottom": 0,
              }
            }
          >
            black
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "justifyContent": "flex-end",
              }
            }
          >
            3
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginRight": 8,
              "maxWidth": 64,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "marginBottom": 0,
              }
            }
          >
            white
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "justifyContent": "flex-end",
              }
            }
          >
            5
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginRight": 8,
              "maxWidth": 64,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "marginBottom": 0,
              }
            }
          >
            burgandy
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "justifyContent": "flex-end",
              }
            }
          >
            4
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginRight": 8,
              "maxWidth": 64,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "marginBottom": 0,
              }
            }
          >
            black & white
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "justifyContent": "flex-end",
              }
            }
          >
            0
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginRight": 8,
              "maxWidth": 64,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "marginBottom": 0,
              }
            }
          >
            black, white, & gray
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "justifyContent": "flex-end",
              }
            }
          >
            4
          </Text>
        </View>
      </View>
      <Text
        style={
          Object {
            "fontFamily": "FanwoodText_400Regular",
          }
        }
      >
        $5,000
      </Text>
    </View>
    <View
      style={Object {}}
    >
      <Image
        source={
          Object {
            "uri": "https://via.placeholder.com/150",
          }
        }
        style={
          Object {
            "height": 223,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "Quattrocento_700Bold",
            "fontSize": 20,
            "fontWeight": "700",
          }
        }
      >
        Chanel
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "FanwoodText_400Regular",
            "fontSize": 20,
          }
        }
      >
        Flap Bag 2
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginRight": 8,
              "maxWidth": 64,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "marginBottom": 0,
              }
            }
          >
            black
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "justifyContent": "flex-end",
              }
            }
          >
            3
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginRight": 8,
              "maxWidth": 64,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "marginBottom": 0,
              }
            }
          >
            white
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "justifyContent": "flex-end",
              }
            }
          >
            5
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginRight": 8,
              "maxWidth": 64,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "marginBottom": 0,
              }
            }
          >
            burgandy
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "justifyContent": "flex-end",
              }
            }
          >
            4
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginRight": 8,
              "maxWidth": 64,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "marginBottom": 0,
              }
            }
          >
            black & white
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "justifyContent": "flex-end",
              }
            }
          >
            0
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginRight": 8,
              "maxWidth": 64,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "marginBottom": 0,
              }
            }
          >
            black, white, & gray
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "FanwoodText_400Regular",
                "justifyContent": "flex-end",
              }
            }
          >
            4
          </Text>
        </View>
      </View>
      <Text
        style={
          Object {
            "fontFamily": "FanwoodText_400Regular",
          }
        }
      >
        $5,000
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders correctly with no data 1`] = `
<Text
  style={
    Object {
      "fontFamily": "Quattrocento_400Regular",
    }
  }
>
  Something went wrong. No products are found yet!
</Text>
`;

exports[`renders correctly with undefined data 1`] = `
<Text
  style={
    Object {
      "fontFamily": "Quattrocento_400Regular",
    }
  }
>
  Something went wrong. No products are found yet!
</Text>
`;
